app/controllers/admin/publishers/referrals_controller.rb

```diff
@@ -0,0 +1,6 @@
+class Admin::Publishers::ReferralsController < Admin::PublishersController
```

Comment:
It's preferable to inherit from the administrator ApplicationController, rather than another models controller. When we inherit from another controller then we pollute our controller class somewhat. In addition all the before actions will be called on this making it somewhat more difficult to effectively determine what's going on and potentially leading to unintended side effects.

Reply:
I don't agree with this, as wouldn't sub routes typically have the same before_action conditions as their parent routes? I'm trying to keep in mind DRY conventions.

That being said, I did fix an error on the next line.